iphone/Maps/Classes/CustomAlert/MWMOsmReauthAlert.mm

```diff
@@ -0,0 +1,69 @@
+#import "MWMAlertViewController.h"
+#import "MWMOsmReauthAlert.h"
+#import "MWMAuthorizationCommon.h"
+
+#include "editor/osm_auth.hpp"
+
+static NSString * const kMap2OsmLoginSegue = @"Map2OsmLogin";
+
+@implementation MWMOsmReauthAlert
+
++ (instancetype)alert
+{
+  MWMOsmReauthAlert * alert =
+      [NSBundle.mainBundle loadNibNamed:[self className] owner:nil options:nil].firstObject;
+
+  alert.messageLabel.userInteractionEnabled = YES;
+  alert.messageLabel.attributedText = [self buildAlertMessage];
+  alert.messageLabel.textAlignment = NSTextAlignmentCenter;
+  UITapGestureRecognizer *tapGesture =
+        [[UITapGestureRecognizer alloc] initWithTarget:alert
+                                                action:@selector(showMoreInfo)];
+  [alert.messageLabel addGestureRecognizer:tapGesture];
+
+  return alert;
+}
+
+// Build attributed string in format "{alert_reauth_message_ios} {alert_reauth_link_text_ios}"
+// where {alert_reauth_link_text_ios} has blue color as a link
++ (NSMutableAttributedString*)buildAlertMessage
+{
+  auto textAttrs = @{NSFontAttributeName : UIFont.regular17};
+  auto linkAttrs = @{NSForegroundColorAttributeName : UIColor.linkBlue,
+                     NSFontAttributeName : UIFont.regular17};
```

Comment:
Usually for links `NSLinkAttributeName` with the URL as `value` to make a tappable link inside the text fully handled by the system.
So the link for the only `here` can be added without explicit link opening.

Reply:
Changed link handling with `NSLinkAttributeName`